### DISCUSSION TITLE
Including Jinja2 templates in other templates

### DISCUSSION THREAD

**Discussion by fatman00** [netbox-community/netbox#13602] [Help Wanted!]
Hi,

I am trying to use the new Templates system in Netbox v3.5.6 and would like to have a base template with some macros for the common configuration I have on all my devices(aaa, ntp, dns, line config).

But I am not able to get the my device template to include the base config.

I have created a Config template called Switch Config that I attache to all my switches but in that I use {% include 'BaseConfig' %} at the top according to: https://github.com/netbox-community/netbox/issues/11559#issuecomment-1426072654

I have a Config Template called BaseConfig as well that includes the configuration I want to include, but when I try to use it the interpreter just gives me an error as shown below:

```
Traceback (most recent call last):
  File "/opt/netbox/netbox/dcim/views.py", line 2075, in get_extra_context
    rendered_config = config_template.render(context=context_data)
  File "/opt/netbox/netbox/extras/models/configs.py", line 252, in render
    output = template.render(**context)
  File "/opt/netbox/venv/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/opt/netbox/venv/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 1, in top-level template code
  File "/opt/netbox/venv/lib/python3.10/site-packages/jinja2/loaders.py", line 99, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: BaseConfig
```

Is there any way I can include some of the other templates I have?

**Comment by kkthxbye-code** [comment]
Including templates like that only works when both files are in the same data source. It does not work at all for manually created config templates.

**Reply by fatman00** [reply]
Hi,

Is there anyway to add this to a feature so all template configuration can be done using the GUI?
Do we have any guide to how to use Github as a datasource for all the templates then?

**Reply by kkthxbye-code** [reply]
> Is there anyway to add this to a feature so all template configuration can be done using the GUI?

You would have to [create a feature request](https://github.com/netbox-community/netbox/blob/develop/CONTRIBUTING.md#bulb-feature-requests).

> Do we have any guide to how to use Github as a datasource for all the templates then?

Not to my knowledge, there's so very light documentation of the general features here:

https://docs.netbox.dev/en/stable/features/configuration-rendering/

https://docs.netbox.dev/en/stable/features/synchronized-data/
